From c755835488ee89a76b945077012847285909745d Mon Sep 17 00:00:00 2001
From: Phil Edworthy <phil.edworthy@renesas.com>
Date: Thu, 24 Nov 2016 16:29:12 +0000
Subject: [PATCH] sf: Don't include erase and write code for SPL

If it wasn't for bank support, we could get rid of all the write code

Signed-off-by: Phil Edworthy <phil.edworthy@renesas.com>
---
 drivers/mtd/spi/sf_probe.c  | 4 ++++
 drivers/mtd/spi/spi_flash.c | 2 ++
 2 files changed, 6 insertions(+)

diff --git a/drivers/mtd/spi/sf_probe.c b/drivers/mtd/spi/sf_probe.c
index 7b29637..676b1df 100644
--- a/drivers/mtd/spi/sf_probe.c
+++ b/drivers/mtd/spi/sf_probe.c
@@ -119,6 +119,7 @@ static int spi_flash_std_read(struct udevice *dev, u32 offset, size_t len,
 	return spi_flash_cmd_read_ops(flash, offset, len, buf);
 }
 
+#if !defined(CONFIG_SPL_BUILD)
 static int spi_flash_std_write(struct udevice *dev, u32 offset, size_t len,
 			const void *buf)
 {
@@ -142,6 +143,7 @@ static int spi_flash_std_erase(struct udevice *dev, u32 offset, size_t len)
 
 	return spi_flash_cmd_erase_ops(flash, offset, len);
 }
+#endif
 
 static int spi_flash_std_probe(struct udevice *dev)
 {
@@ -158,8 +160,10 @@ static int spi_flash_std_probe(struct udevice *dev)
 
 static const struct dm_spi_flash_ops spi_flash_std_ops = {
 	.read = spi_flash_std_read,
+#if !defined(CONFIG_SPL_BUILD)
 	.write = spi_flash_std_write,
 	.erase = spi_flash_std_erase,
+#endif
 };
 
 static const struct udevice_id spi_flash_std_ids[] = {
diff --git a/drivers/mtd/spi/spi_flash.c b/drivers/mtd/spi/spi_flash.c
index c529f38..fb3a3f4 100644
--- a/drivers/mtd/spi/spi_flash.c
+++ b/drivers/mtd/spi/spi_flash.c
@@ -328,6 +328,7 @@ int spi_flash_write_common(struct spi_flash *flash, const u8 *cmd,
 	return ret;
 }
 
+#if !defined(CONFIG_SPL_BUILD)
 int spi_flash_cmd_erase_ops(struct spi_flash *flash, u32 offset, size_t len)
 {
 	u32 erase_size, erase_addr;
@@ -469,6 +470,7 @@ int spi_flash_cmd_write_ops(struct spi_flash *flash, u32 offset,
 
 	return ret;
 }
+#endif /* !defined(CONFIG_SPL_BUILD) */
 
 int spi_flash_read_common(struct spi_flash *flash, const u8 *cmd,
 		size_t cmd_len, void *data, size_t data_len)
-- 
2.7.4

